### DISCUSSION TITLE
High poly situations

### DISCUSSION THREAD

**Discussion by unicomp21** [zeux/meshoptimizer#279] [Ideas]
What's the best path for baking high poly assets which have already been processed by mesh optimizer?  Also, is there a way to compress vertex colors?

**Comment by zeux** [comment]
What do you mean by "baking"? Vertex colors are compressed by default (using quantization + `-cc` option applies further compression as with other attributes).

**Comment by unicomp21** [comment]
To lossy compress things further, this is the sort of thing I'm talking about.  I'm thinking if we could have an auto uv unwrap into quads which then ray cast the texture tiles against the mesh to get normals, color, etc.  Would be a nice feature.

https://marmoset.co/posts/toolbag-baking-tutorial/

I realize meshlets are the future, but webgpu still has a ways to go.

**Comment by AndrewJDR** [comment]
@unicomp21 It seems like you're looking for attribute aware simplification, like https://github.com/zeux/meshoptimizer/issues/158 ? (Not in yet, but planned...)

**Comment by unicomp21** [comment]
@AndrewJDR looks to be related, but would be nice if the cages and uv unwrap (or meshlet generation) could be automated?  After that stage, if we could have something for baking textures that would be cool too.  Perhaps it would be a different project, once the cages and uv unwrap are automated.  Or, maybe I'm in left field, lol.

**Comment by zeux** [comment]
Yeah I think this is talking about a remeshing approach where also instead of using the original texture, a new texture should be synthesized to closely mirror the original appearance. I had remeshing on my todo list, but I don't have immediate plans to implement it - need to finish other projects first. I'm going to convert this to a discussion thread so that the conversation can continue.
